lib/wakes/matchers.rb

```diff
@@ -1,20 +1,42 @@
 # frozen_string_literal: true
+
 RSpec::Matchers.define :have_wakes_graph do |canonical_location:, legacy_locations: []|
+  def host(location_string)
+    location_string = "http://#{location_string}" if URI(location_string).scheme.nil?
```

Comment:
specifically, 

    def uri(location_string)
      location_string = "http://#{location_string}" if URI(location_string).scheme.nil?
      URI(location_string)
    end

Then `#host` and `#path` just need to call `uri(location_string).path` and `uri(location_string).host`